app/pulp/app/models/task.py

```diff
@@ -35,6 +35,30 @@ class ReservedResource(Model):
     worker = models.ForeignKey("Worker", on_delete=models.CASCADE, related_name="reservations")
 
 
+class WorkerManager(models.Manager):
+    def get_unreserved_worker(self):
+        """
+        Find an unreserved :class:`~pulp.app.models.Worker`
+
+        Return the Worker instance that has no :class:`~pulp.app.models.ReservedResource`
+        associated with it. If all workers have ReservedResource relationships a,
+        :class:`pulp.app.models.Worker.DoesNotExist` exception is raised.
+
+        This function also provides randomization for worker selection.
+
+        :raises Worker.DoesNotExist: If all workers have ReservedResource entries associated with
+                                     them.
+
+        :returns:          The Worker instance that has no reserved_resource
```

Comment:
s/reserved_resource/ReservedResource/
Also consider replacing with `A random Worker instance that has no associated ReservedReservation entries`.